### PR TITLE
fix(pipelineTriggers): change buildNumber type from int to string

### PR DESCRIPTION
**Pull request by jaredstehler**
in order to support build numbers which are not strict integers (i.e. "123.1")

https://github.com/spinnaker/spinnaker/issues/6743

### CODE REVIEW

File: orca-bakery/src/test/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTaskSpec.groovy [hunk 1]
```diff
@@ -118,7 +118,7 @@ class CreateBakeTaskSpec extends Specification {
 
   @Shared
   def buildInfo = new JenkinsBuildInfo(
-    "name", 0, "http://jenkins", "SUCCESS",
+    "name", null, "http://jenkins", "SUCCESS",
```
Comment (dbyron-sf) [inline orca-bakery/src/test/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTaskSpec.groovy:121]: Wouldn't `"0"` be a more straightforward conversion?  Though I'm all for testing with null too.

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/BuildInfo.kt [hunk 2]
```diff
@@ -2,7 +2,7 @@ package com.netflix.spinnaker.orca.pipeline.model
 
 abstract class BuildInfo<A>(
   open val name: String?,
-  open val number: Int,
+  open val number: String?,
```
Comment (dbyron-sf) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/BuildInfo.kt:5]: Why nullable when it wasn't before?

Comment (jaredstehler) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/BuildInfo.kt:5]: i believe in this case the constant `-1` was being sent to indicate null value

**Comment by dbyron-sf** [comment]
@mergifyio update

File: orca-bakery/src/test/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTaskSpec.groovy (hunk 1 above)
Comment (jaredstehler) [reply orca-bakery/src/test/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTaskSpec.groovy:121]: i could refactor to treat `"0"` as null/missing?

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/BuildDetailExtractor.java [hunk 3]
```diff
@@ -112,7 +112,9 @@ public boolean tryToExtractBuildDetails(BuildInfo<?> buildInfo, Map<String, Obje
       if (buildInfo == null || request == null) {
         return false;
       }
-      if (buildInfo.getUrl() != null && buildInfo.getName() != null && buildInfo.getNumber() > 0) {
```
Comment (dbyron-sf) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/BuildDetailExtractor.java:115]: Aaah, here's the place that was treating a number <= 0 as missing...

File: orca-bakery/src/test/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTaskSpec.groovy (hunk 1 above)
Comment (dbyron-sf) [reply orca-bakery/src/test/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTaskSpec.groovy:121]: I think it's fine to use null to indicate missing...but then [this comment](https://github.com/spinnaker/orca/pull/4316#discussion_r1005038701) says -1 used to mean missing....but from [here](https://github.com/spinnaker/orca/pull/4316/files#r1020612013) and specifically the `buildInfo.getNumber() > 0` part...it's not only -1, is everything <= 0 that's considered missing...so this is fine.  Thanks for reading :)

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/BuildInfo.kt (hunk 2 above)
Comment (dbyron-sf) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/BuildInfo.kt:5]: Found [this place](https://github.com/spinnaker/orca/pull/4316/files#r1020612013) that treats <= 0 as missing....`String?` works for me.

**Comment by jaredstehler** [comment]
can this be merged yet?
